src/manageiq_client/api.py

```diff
@@ -587,7 +587,7 @@ def collection(self):
 
     def __getattr__(self, attr):
         self.reload()
-        if not hasattr(self, attr):
+        if attr not in self.__dict__:
             raise AttributeError("No such action {}".format(attr))
         return getattr(self, attr)
```

Comment:
consider rewriting this line too as `self.__dict__[attr]` to explicitly avoid recursion.
(not really necessary, getattr is not supposed to call __getattr__ if the attribute exists, just for readabily)

Reply:
+1 I agree. I will make the requested change and update the PR.